tutorials/xr/ar_passthrough.rst

```diff
@@ -0,0 +1,187 @@
+
+Headsets such as the Magic Leap and glasses such as TiltFive show the rendered result on transparent displays
+allowing the user to see the real world.
+
+Headsets such as the Quest, HTC Elite, and Lynx R1 implement this through a technique called passthrough,
+where cameras record the real world and these images are used as the background on top of which our rendered
+result is used.
+
+.. note::
+
+    The way specifically passthrough is implemented has vast differences between different platforms.
+
+    In Godot 4.3 we have implemented a unified approach that is explained on this help page
+    so you don't need to worry about these differences, the :ref:`XRInterface <class_xrinterface>`
+    implementation is now responsible for applying the correct platform dependent method [#]_.
+
+    For headsets such as the Meta Quest and HTC Elite you will need to use the
+    `OpenXR vendors plugin v3.0.0 <https://github.com/GodotVR/godot_openxr_vendors/releases>`__
+    or later to enable passthrough.
+
+    For backwards compatibility the old API for passthrough is still available but it is recommended
+    to follow the new instructions below.
+
+Environment blend modes
+-----------------------
+
+The way we configure VR or AR functionality is through setting the environment blend mode.
+This mode determines how the (real world) environment is blended with the virtual world.
+
+.. list-table:: Blend modes
+  :widths: 35 65
+  :header-rows: 1
+
+  * - Blend mode
+    - Description
+  * - XR_ENV_BLEND_MODE_OPAQUE
+    - The rendered image is opaque, we do not see the real world. We're in VR mode.
+      This will turn off passthrough if passthrough is used.
+  * - XR_ENV_BLEND_MODE_ADDITIVE
+    - The rendered image is added to the real world. This mode generally applies to AR only devices
+      such as the Hololens where the optics are not able to obscure the real world.
+      This will turn on passthrough if passthrough is used.
+  * - XR_ENV_BLEND_MODE_ALPHA_BLEND
+    - The rendered image is alpha blended with the real world.
+      This is supported on devices that can fully obscure the real world
+      and generally the preferred way to do AR.
+      This will turn on passthrough if passthrough is used.
+  
+You can set the environment blend mode for your application through the ``environment_blend_mode``
+property of the :ref:`XRInterface <class_xrinterface>` instance.
+
+You can query the supported blend modes on the hardware using the
+``get_supported_environment_blend_modes`` property on the same instance.
+
+Configuring your background
+---------------------------
+
+When setting the blend mode to ``XR_ENV_BLEND_MODE_ALPHA_BLEND`` you must set
+the ``transparent_bg`` property on :ref:`Viewport <class_viewport>` to true.
+It is advisable to do the same with ``XR_ENV_BLEND_MODE_ADDITIVE`` however here you
+can alternatively set your background color to black.
```

Comment:
This seems to assume that `XR_ENV_BLEND_MODE_ADDITIVE` is only used on see-through devices (otherwise setting the background color to black would hide the real world on video-passthrough devices) but we don't mention it so in the section above.

Reply:
Additive will **add** the color, and as with black you are adding zeros to whatever is underneath, you'll just see the real world. So if this is implemented properly on video passthrough, you should still see the background, it should **not** turn the background black.
Alpha in this case is ignored and in there lies a minor danger if you're relying on alpha as colors will actually be brighter than you expect.

Actually thinking about this now, I think I should say exactly that here and just recommend the user sets the background to black, and leave it there.

In reality, a device reporting whether it supports additive, or alpha, or both, is what informs you about the limitation of the AR technology being used and whether you as a developer want to do something with that (in many cases you don't and just accept the limitation).

Reply:
Ok, changed this, let me know what you think.